zhaquirks/nodon/pilot_wire.py

```diff
@@ -0,0 +1,100 @@
+"""NodOn pilot wire heating module."""
+
+from zigpy.quirks import CustomCluster
+from zigpy.quirks.v2 import EntityType, QuirkBuilder
+import zigpy.types as t
+from zigpy.zcl.foundation import (
+    BaseAttributeDefs,
+    BaseCommandDefs,
+    DataTypeId,
+    Direction,
+    ZCLAttributeDef,
+    ZCLCommandDef,
+)
+
+NODON = "NodOn"
+NODON_MANUFACTURER_ID = 4747
+NODON_PILOT_WIRE_CLUSTER_ID = 0xFC00  # 64512
+ADEO = "Adeo"
+
+
+class NodOnPilotWireMode(t.enum8):
+    """Pilot wire mode."""
+
+    # Codes taken from
+    # https://github.com/Koenkk/zigbee-herdsman-converters/blob/0f4833340a20db3dae625a61c41d9be0a6f952be/src/converters/fromZigbee.ts#L5285.
+
+    Off = 0x00
+    Comfort = 0x01
+    Eco = 0x02
+    FrostProtection = 0x03
+    ComfortMinus1 = 0x04
+    ComfortMinus2 = 0x05
+
+
+class BasePilotWireCluster(CustomCluster):
+    """Base cluster to set Pilot Wire mode."""
+
+    name: str = "PilotWireCluster"
+    cluster_id: t.uint16_t = NODON_PILOT_WIRE_CLUSTER_ID
+    ep_attribute: str = "pilot_wire_cluster"
+
+    class AttributeDefs(BaseAttributeDefs):
+        """Attribute definitions."""
+
+        pilot_wire_mode = ZCLAttributeDef(
+            id=0x0000,
+            type=NodOnPilotWireMode,
+            # I got the following error without setting zcl_type explicitly to int:
+            # Failed to write attribute pilot_wire_mode=<NodOnPilotWireMode.FrostProtection: 3>: <Status.INVALID_DATA_TYPE: 141>
+            zcl_type=DataTypeId.uint8,
+            is_manufacturer_specific=True,
+        )
+
+    class ServerCommandDefs(BaseCommandDefs):
+        """Server command definitions."""
+
+        set_pilot_wire_mode = ZCLCommandDef(
+            id=0x00,
+            schema={"mode": NodOnPilotWireMode},
+            direction=Direction.Client_to_Server,
+            is_manufacturer_specific=True,
+        )
```

Comment:
This command is now unused, right? Is there a difference to directly setting the `pilot_wire_mode` attribute vs using the command?

Reply:
ahm... yes, it's unused. I'll remove it

Reply:
You can leave it in if it's present on the device. Might be useful for debugging or something. It can be access through the "Manage Zigbee device" UI.
I was just curious if there's a difference to using that vs setting the attribute directly.